File: app/src/main/java/com/alexkn/syntact/presentation/bucketlist/PlayMenuViewModel.java
package com.alexkn.syntact.presentation.bucketlist;

import android.os.AsyncTask;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.alexkn.syntact.data.model.Bucket;
import com.alexkn.syntact.data.model.views.BucketDetail;
import com.alexkn.syntact.domain.repository.BucketRepository;

import java.util.List;

import javax.inject.Inject;

public class PlayMenuViewModel extends ViewModel {

    private BucketRepository bucketRepository;

    private LiveData<List<BucketDetail>> buckets;

    @Inject
    public PlayMenuViewModel(BucketRepository bucketRepository) {

        super();
        this.bucketRepository = bucketRepository;
        buckets = bucketRepository.getBucketDetails();
    }

    public void deleteLanguage(Bucket bucket) {

        AsyncTask.execute(() -> bucketRepository.removeLanguage(bucket));
    }

    LiveData<List<BucketDetail>> getBuckets() {

        return buckets;
    }
}

File: app/src/main/java/com/alexkn/syntact/presentation/bucketlist/PlayMenuViewModel.kt
package com.alexkn.syntact.presentation.bucketlist

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope

import com.alexkn.syntact.data.model.views.BucketDetail
import com.alexkn.syntact.data.model.views.PlayerStats
import com.alexkn.syntact.core.repository.BucketRepository
import kotlinx.coroutines.launch

import javax.inject.Inject

class PlayMenuViewModel @Inject
constructor(private val bucketRepository: BucketRepository) : ViewModel() {

    val buckets: LiveData<List<BucketDetail>> = bucketRepository.bucketDetails

    val playerStats: LiveData<PlayerStats> = bucketRepository.getPlayerStats()

    fun deleteBucket(id: Long) {
        viewModelScope.launch { bucketRepository.deleteBucket(id) }
    }
}

